### PR TITLE
feat(app): integrate existing AnomalyGenNext synthesis

### PR DESCRIPTION
**Pull request by echen4628**
### What changes are proposed in this pull request?

This PR adds an optional AnomalyGenNext synthesis route to the RT-DETR DEFT OD AOI application for datasets that already have compatible AnomalyGenNext task weights.

The new `prepare_deft_od_aoi_synthesis.py` helper:

- reads strict gap-analysis output and keeps false negatives only;
- requires each false-negative box to match a frozen KPI COCO annotation exactly;
- resolves explicit `dataset_id`, `texture_id`, `defect_class`, and pixel `fn_mask_source` metadata from the matched KPI records;
- rejects boxes without real pixel masks, unknown dataset routes, missing source files, and mismatched frozen KPI paths;
- emits normalized FN gaps and a filtering YAML for `tao-prepare-anomalygennext-inputs`; and
- carries the frozen SigLIP identity, retrieval thresholds, AnomalyGenNext checkpoint/recipe routes, defect specification, and AMP settings into the handoff.

The application initializer now validates synthesis prerequisites when synthesis is enabled and records whether the run uses `rtdetr_real_only` or `rtdetr_with_synthesis`. The durable state machine inserts `iteration_synthesis` between admission and training only for synthesis-enabled runs.

The cumulative COCO admission helper now accepts a generated COCO document plus its image root. It folds generated anomaly categories into the application's single `defect` category, preserves synthetic provenance, deduplicates sources, and enforces the frozen cumulative synthetic-to-real defect cap. Supplying only one of the synthetic COCO/image-root pair is rejected.

Synthesis remains disabled by default:

```yaml
synthesis:
  enabled: true
  pool_dataset_root: /data/anomalygen_pool
  defect_spec: /data/defect_spec.jsonl
  routes:
    line_a:
      checkpoint: /models/line_a/adapter.pt
      recipe: /models/line_a/canonical_recipe.yaml
```

After strict gap analysis:

```bash
skills/applications/tao-run-deft-od-aoi/scripts/prepare_deft_od_aoi_synthesis.py \
  --policy /results/deft_od_aoi_policy.yaml \
  --strict-gaps /results/iteration_1/measure/gap_strict/box_gaps.parquet \
  --output-dir /results/iteration_1/synthesis_request
```

The emitted filtering YAML is consumed by the existing `tao-prepare-anomalygennext-inputs` and `tao-generate-od-defects` leaves. This PR supports existing task weights only; a following stacked PR adds a one-time fine-tuning bootstrap for routes without them.

### Why are the changes needed?

Real-data mining remains the safest default, but some AOI datasets benefit from generating additional examples for strict false negatives. The application needs a narrow integration boundary that preserves frozen KPI identity and provenance while preventing detector boxes from being treated as segmentation masks.

This route makes synthesis opt-in and policy-bound. It reuses the dedicated AnomalyGenNext preparation and generation leaves, requires existing compatible checkpoint/recipe pairs, and admits only validated outputs under a cumulative cap. That keeps generation separate from detector training and prevents synthetic data from growing without reference to admitted real defects.

### Related issues

N/A

### Does this PR introduce _any_ user-facing change?

Yes.

Previously, `tao-run-deft-od-aoi` supported only the real-data RT-DETR loop.

After this change, users with existing AnomalyGenNext task weights can enable an optional synthesis stage. They must provide a normalized reference pool, defect specification, per-dataset checkpoint/recipe routes, and exact synthesis metadata plus pixel masks on eligible KPI records. Generated samples can then be admitted into the cumulative binary detector dataset under a configured fraction cap.

The existing real-only workflow is unchanged when `synthesis.enabled` remains `false`; no migration is required.

### How was this patch tested?

The branch-focused synthesis tests were run against this stacked slice:

```text
$ .venv/deft/bin/python -m pytest -q \
    skills/applications/tao-run-deft-od-aoi/scripts/tests/test_init_deft_od_aoi.py \
    skills/applications/tao-run-deft-od-aoi/scripts/tests/test_admit_deft_od_aoi_coco.py \
    skills/applications/tao-run-deft-od-aoi/scripts/tests/test_commit_deft_od_aoi_stage.py \
    skills/applications/tao-run-deft-od-aoi/scripts/tests/test_prepare_deft_od_aoi_synthesis.py

10 passed in 0.58s
```

The tests cover:

- normalizing an exact strict KPI false negative into an AnomalyGenNext route;
- preserving the explicit pixel-mask path and derived anomaly type;
- validating existing checkpoint/recipe routes during initialization;
- requiring the synthesis stage only when synthesis is enabled;
- keeping the real-only stage graph unchanged when it is disabled;
- admitting generated images under the cumulative cap;
- folding generated fine-grained categories to the binary `defect` category;
- preserving explicit zero-annotation clean images;
- rejecting an empty real/clean admission after the similarity gate; and
- rejecting incomplete synthetic COCO/image-root input pairs.

The skill validator and patch-integrity check also passed:

```text
$ VALIDATE_BASE_REF=dev/deft-od-aoi-rtdetr-selection ./scripts/validate-skills.sh

✓ validate-skills passed

$ git diff --check dev/deft-od-aoi-rtdetr-selection...dev/deft-od-aoi-synthesis-existing

# no output
```

The downstream AnomalyGenNext contracts used by this integration were also exercised with the public AnomalyGenNext 1.1 container on one H100. Two explicit false negatives produced six retrieval candidates, 12 successful native AMP rows, two selected pairs, and four generation rows. Generation then completed all four requested outputs with four native annotations and valid fine-grained and binary COCO documents. No guardrail-disable flag was used; text screening and face blurring ran, while the public preset reported that an enforcing image-safety model was unavailable.

### Was this patch authored or co-authored using generative AI tooling?

Yes. This patch was co-authored using OpenAI Codex.

Generated-by: OpenAI Codex (GPT-5.6)

### Release note

```release-note
Added optional AnomalyGenNext synthesis for DEFT OD AOI datasets with existing task checkpoints and recipes.
```

### Checklist

- [x] My commits are signed off per the DCO (`git commit -s`) — see `CONTRIBUTING.md`
- [x] I have read the contributing guidelines
- [x] The code follows the project's style, and lint and format checks pass locally
- [x] I added or updated tests covering this change
- [x] All tests pass locally
- [x] I added or updated documentation (README, docstrings, docs pages, examples)
- [x] I updated the version, changelog, or migration notes if this change requires it
- [x] If this touches components that are optional to install, the imports are guarded
      so the package still works without them (reviewers: please verify this)
- [x] No secrets, credentials, proprietary data, or customer data are included in this PR
- [x] I understand this contribution is licensed under the repository's license, and that
      my commit author name and email become permanently public once merged

### Notes for reviewers

The most useful review order is:

1. `references/anomalygen-pool.md` and the new `synthesis` policy block for the public contract.
2. `scripts/prepare_deft_od_aoi_synthesis.py` for strict-FN normalization and validation.
3. The synthesis additions in `scripts/admit_deft_od_aoi_coco.py` for capped binary admission.
4. The initializer and stage-commit changes for the opt-in state transition.

The route deliberately requires an exact frozen KPI annotation match and an existing pixel mask. It does not infer dataset identity, anomaly type, or masks from filenames or boxes.

This PR accepts only existing checkpoint/recipe pairs. Missing AnomalyGenNext task weights are handled separately by the following `dev/deft-od-aoi-synthesis-bootstrap` PR so training occurs once before the DEFT loop rather than inside an iteration.

### CODE REVIEW

File: versions.yaml [hunk 1]
```diff
@@ -47,6 +47,7 @@ images:
 
   metropolis_sdg:
     paidf_anomalygen: nvcr.io/nvidia/paidf-anomalygen:1.0.1
+    anomalygen_next: nvcr.io/nvidia/paidf-anomalygen:1.1.0
```
Comment (xiaosean) [inline versions.yaml:50]: Suggest to use **paidf_anomalygen_cosmos3**

File: skills/data/tao-finetune-anomalygennext/references/container-runtime.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Container runtime
+
+Read this before submitting AnomalyGenNext fine-tuning.
+
+The action uses the public image declared by `skill_info.yaml`:
+
+```text
+nvcr.io/nvidia/paidf-anomalygen:1.1.0
```
Comment (xiaosean) [inline skills/data/tao-finetune-anomalygennext/references/container-runtime.md:8]: Small suggestion: would it make sense to have this point to the exact registry entry in versions.yaml instead of hardcoding it in the skill?

**Review by xiaosean — APPROVED** [review]
